src/main/java/com/squareup/kotlinpoet/CodeBlock.kt

```diff
@@ -379,3 +379,8 @@ class CodeBlock private constructor(
         = c == '%' || c == '>' || c == '<' || c == '[' || c == ']' || c == 'W'
   }
 }
+
+@JvmOverloads
+fun Iterable<CodeBlock>.joinToCode(separator: CharSequence = ", ", prefix: CharSequence = "",
+                                   suffix: CharSequence = "" ) =
+    CodeBlock.of(map { it.toString() }.joinToString(separator, prefix, suffix))
```

Comment:
`toTypedArray` is available for `Collection` but not for `Iterable`. I've changed extension type to `Collection` for now. If it is needed I can do revert it back to `Iterable` and convert to array manually.